extensions/pkg/controller/infrastructure/actuator.go

```diff
@@ -13,16 +13,61 @@ import (
 	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
 )
 
-// Actuator acts upon Infrastructure resources.
+// Actuator acts upon [extensionsv1alpha1.Infrastructure] resources.
 type Actuator interface {
-	// Reconcile the Infrastructure config.
+	// Reconcile reconciles the [extensionsv1alpha1.Infrastructure] resource.
+	//
+	// Implementations should ensure that required infrastructure resources
+	// are created or updated to reach the desired state.
 	Reconcile(context.Context, logr.Logger, *extensionsv1alpha1.Infrastructure, *extensionscontroller.Cluster) error
-	// Delete the Infrastructure config.
+
+	// Delete is invoked when the [extensionsv1alpha1.Infrastructure]
+	// resource is deleted in a graceful manner.
+	//
+	// Implementations should take care of cleaning up any infrastructure
+	// related resources (e.g. virtual machines) which were created by the
+	// Extension.
+	//
+	// Implementations must wait until all resources managed by the
+	// extension have been gracefully cleaned up.
 	Delete(context.Context, logr.Logger, *extensionsv1alpha1.Infrastructure, *extensionscontroller.Cluster) error
-	// ForceDelete forcefully deletes the Infrastructure config.
+
+	// ForceDelete is invoked when the shoot cluster associated with the
+	// [extensionsv1alpha1.Infrastructure] resource is being deleted in a
+	// forceful manner.
+	//
+	// Implementations should take care of unblocking the deletion flow by
+	// attempting to cleanup any resources created by the extension, remove
+	// any finalizers created for custom resources, etc., and also skip
+	// waiting for external resources (e.g. virtual machines), if they
```

Comment:
```suggestion
	// waiting for external resources (e.g. custom routes or subnets), if they
```

Nit, but reflects the usual case.